pajbot/modules/leaguerank.py

```diff
@@ -129,44 +142,37 @@ def league_rank(self, bot, source, message, **rest):
         if len(summoner_name) == 0 or len(region) == 0:
             return False
 
-        error_404 = "Game data not found"
-        error_429 = "Too many requests"
-
         try:
-            rw = RiotWatcher(riot_api_key, default_region=region)
+            lw = LolWatcher(riot_api_key)
 
-            summoner = rw.get_summoner(name=summoner_name)
+            summoner = lw.summoner.by_name(region, summoner_name)
             summoner_id = str(summoner["id"])
             summoner_name = summoner["name"]
 
-        except LoLException as e:
-            if e == error_429:
+        except ApiError as e:
+            log.exception("babyrage")
+            if e.response.status_code == 429:
                 bot.say(f"Too many requests. Try again in {e.headers['Retry-After']} seconds")
-                return False
-            elif e == error_404:
+            elif e.response.status_code == 404:
                 bot.say("The summoner not found. Use a valid summoner name (remove spaces) and region FailFish")
-                return False
-            else:
-                log.info(f"Something unknown went wrong: {e}")
-                return False
+            return False
 
         try:
-            summoner_league = rw.get_league_entry(summoner_ids=(summoner_id,))
+            summoner_league = lw.league.by_summoner(region, summoner_id)
 
-            tier = summoner_league[summoner_id][0]["tier"]
-            division = summoner_league[summoner_id][0]["entries"][0]["division"]
-            league_points = summoner_league[summoner_id][0]["entries"][0]["leaguePoints"]
+            tier = summoner_league[0]["tier"]
+            rank = summoner_league[0]["rank"]
+            league_points = summoner_league[0]["leaguePoints"]
 
             bot.say(
-                f"The Summoner {summoner_name} on region {region.upper()} is currently in {tier} {division} with {league_points} LP 4Head"
+                f"The Summoner {summoner_name} on region {region.upper()} is currently in {tier} {rank} with {league_points} LP 4Head"
             )
-        except LoLException as e:
-            if e == error_429:
+        except ApiError as e:
+            log.exception("babyrage")
+            if e.response.status_code == 429:
                 bot.say(f"Too many requests. Try again in {e.headers['Retry-After']} seconds")
```

Comment:
now that I see it, this will have to probably be e.response.headers

Reply:
![image](https://user-images.githubusercontent.com/12804673/90764130-3590c380-e32b-11ea-9f9a-95c9702830ce.png)